src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -0,0 +1,279 @@
+
+/**
+ * Encryption utility class
+ */
+public class EncryptorUtils {
+
+    private static final Logger logger = LogManager.getLogger(EncryptorUtils.class);
+
+    private ClusterService clusterService;
+    private Client client;
+    private String masterKey;
+
+    /**
+     * Instantiates a new EncryptorUtils object
+     * @param clusterService the cluster service
+     * @param client the node client
+     */
+    public EncryptorUtils(ClusterService clusterService, Client client) {
+        this.masterKey = null;
+        this.clusterService = clusterService;
+        this.client = client;
+    }
+
+    /**
+     * Sets the master key
+     * @param masterKey the master key
+     */
+    protected void setMasterKey(String masterKey) {
+        this.masterKey = masterKey;
+    }
+
+    /**
+     * Returns the master key
+     * @return the master key
+     */
+    protected String getMasterKey() {
+        return this.masterKey;
+    }
+
+    /**
+     * Randomly generate a master key
+     * @return the master key string
+     */
+    protected String generateMasterKey() {
+        byte[] masterKeyBytes = new byte[32];
+        new SecureRandom().nextBytes(masterKeyBytes);
+        return Base64.getEncoder().encodeToString(masterKeyBytes);
+    }
+
+    /**
+     * Encrypts template credentials
+     * @param template the template to encrypt
+     * @return template with encrypted credentials
+     */
+    public Template encryptTemplateCredentials(Template template) {
+        return processTemplateCredentials(template, this::encrypt);
+    }
+
+    /**
+     * Decrypts template credentials
+     * @param template the template to decrypt
+     * @return template with decrypted credentials
+     */
+    public Template decryptTemplateCredentials(Template template) {
+        return processTemplateCredentials(template, this::decrypt);
+    }
+
+    /**
+     * Applies the given cipher function on template credentials
+     * @param template the template to process
+     * @param cipher the encryption/decryption function to apply on credential values
+     * @return template with encrypted credentials
+     */
+    private Template processTemplateCredentials(Template template, Function<String, String> cipherFunction) {
+        Template.Builder processedTemplateBuilder = new Template.Builder();
+
+        Map<String, Workflow> processedWorkflows = new HashMap<>();
+        for (Map.Entry<String, Workflow> entry : template.workflows().entrySet()) {
+
+            List<WorkflowNode> processedNodes = new ArrayList<>();
+            for (WorkflowNode node : entry.getValue().nodes()) {
+                if (node.userInputs().containsKey(CREDENTIAL_FIELD)) {
+                    // Apply the cipher funcion on all values within credential field
+                    @SuppressWarnings("unchecked")
+                    Map<String, String> credentials = new HashMap<>((Map<String, String>) node.userInputs().get(CREDENTIAL_FIELD));
```

Comment:
I understand that for now we have the `credential` field as the only field where we want to encrypt and right now it looks like the standard is that whenever this credential field exists in the template we will encrypt it. Do you think we should maybe have a mechanism that is more step specific. For example in `workflow-step.json` we can add a `fields-to-encrypt` list for that step, so we only encrypt on create connector or register model. 

reason for this is because for example if we integrate with creating a data source connection form Promotheus: https://opensearch.org/docs/latest/observing-your-data/prometheusmetrics/
the encrypted fields are: `prometheus.auth.secret_key` and `prometheus.auth.access_key`.  For `personalize_search_ranking` search-pipeline processor the field is just `iam_role_arn`. 

However since it might be a long time until we integrate these it might not be as priority right now.

Reply:
This is a good issue to raise. I think definitely as we encounter future workflow steps that require encryption for different fields, we can proceed on this path, but currently we only need to worry about the `CreateConnectorStep`. As for the `RegisterModelStep`, we currently only support adding the connector ID, not an inline connector, so we don't have any credentials to encrypt for that step. 

I'll go ahead and add a TODO here though

Reply:
yeah good point, just made an issue for the internal connector in registerModel but no worry now. Sounds good on having a TODO and a potential issue on revisiting general encryption logic if we do it based on a list of field names or based on specific steps. WE would need to add in our documentation that every payload with a `credential` field will be encrypted, no other APIs that we have use it though like you mention so its okay. Incase we add an api with a `credential` field that we don't want to encrypt